### PR TITLE
feat: Add response filtering for meta data

### LINKED ISSUE #2308: [FEATURE] Add response filtering for meta data
### Is your feature request related to a problem? Please describe.

Currently the user is not able to filter responses by metadata provided in the responses.

### Describe the solution you'd like

image

image

Let's handle meta data like a multiple choice single question:

Screenshot 2024-03-21 at 12 24 27

### Describe alternatives you've considered

-

### Additional context

-

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
Add response filtering for meta data

Fixes #2308 

## How should this be tested?

- Filter the responses based on the meta data fields

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by Dhruwang — CHANGES_REQUESTED** [review]
Hey @gupta-piyush19 , looks good and filtering works super well 😊🚀

Here are some minor things that I noticed 

1. Double scroll bars

https://github.com/formbricks/formbricks/assets/67850763/15e9e5b7-e6eb-4022-8296-b8bf9ae5b083

2. Can we also change icon for action as # is already used for tags 
Screenshot 2024-04-08 at 2 14 05 PM

3. Here can we rename the section for filtering languages from filter to something else (for example: Other filters) ?
Screenshot 2024-04-08 at 2 15 54 PM

4. Build errors 
Screenshot 2024-04-08 at 2 33 34 PM

Rest looks good ! 😊

**Review by Dhruwang — APPROVED** [review]
Good to go ! 🚀😊

File: packages/lib/response/service.ts [hunk 1]
```diff
@@ -457,6 +458,57 @@ export const getResponsePersonAttributes = async (surveyId: string): Promise<TSu
   return responses;
 };
 
+export const getResponseMeta = async (surveyId: string): Promise<TSurveyMetaFieldFilter> => {
```
Comment (mattinannt) [inline packages/lib/response/service.ts:461]: This service is not cached. Please add caching :-)

**Review by mattinannt — CHANGES_REQUESTED** [review]
@gupta-piyush19 thanks; looks good overall :-) 
Can you please check why the e2e tests are failing and also add caching to the service :-)

**Comment by gupta-piyush19** [comment]
Done with the changes, @mattinannt.
cc: @Dhruwang

**Review by mattinannt — APPROVED** [review]

**Pull request merged**